### PR TITLE
progress bar seek(thumb/slider) feature added

### PR DESCRIPTION
**Pull request by ONyzhnyk-Luxoft**
thumb/slider was added in progress bar 
according with Proposal : Progress Bar Seek Feature

### CODE REVIEW

File: app/controller/sdl/Abstract/Controller.js [hunk 1]
```diff
@@ -196,17 +196,54 @@ SDL.SDLController = Em.Object.extend(
...
-          this.model.progressBar.remove();
-        }
+      // if (this.model) {
+      //   if(this.model.seekBar) {
+          // this.model.progressBar.remove();
+        // }
         SDL.SDLModel.onDeactivateApp(SDL.States.nextState, this.model.appID);
-      }
+      // }
       SDL.SDLController.onSubMenu('top');
```
Comment (ValeriiMalkov) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft Remove comented code

File: app/model/sdl/Abstract/AppModel.js [hunk 2]
```diff
@@ -94,6 +94,8 @@ SDL.ABSAppModel = Em.Object.extend(
      * @type {Number}
      */
     appID: null,
+    valueOfSeekBar: null,
+    style: "",
     /**
```
Comment (ValeriiMalkov) [inline app/model/sdl/Abstract/AppModel.js]: @ONyzhnyk-Luxoft Add description for this values

File: app/model/sdl/MediaModel.js [hunk 3]
```diff
@@ -361,6 +300,7 @@ SDL.SDLMediaModel = SDL.ABSAppModel.extend({
           );
           if(params.enableSeek !== undefined){
             this.set('seekBar', params.enableSeek);
+            
           }
```
Comment (ValeriiMalkov) [inline app/model/sdl/MediaModel.js]: @ONyzhnyk-Luxoft Remove empty line

File: app/view/media/sdl/controllsView.js [hunk 4]
```diff
@@ -56,6 +56,12 @@ SDL.SDLMediaControlls = Em.ContainerView.create(
             '<div class="album textLimit" {{bindAttr style="SDL.SDLController.model.appInfo.alignment"}}>{{SDL.SDLController.model.appInfo.field2}}</div>' +
             '<div class="artist textLimit"{{bindAttr class="SDL.SDLController.model.mediaPreset:hidden"}}>{{SDL.SDLController.model.appInfo.field3}}</div>' +
             '<div class="time"{{bindAttr class="SDL.SDLController.model.mediaPreset:timeV2"}}>{{SDL.SDLController.model.appInfo.mediaClock}}</div>' +
+            '<input type="range" min="0" max="100"' + 
+            'onchange="SDL.SDLController.seekTracking(event)" ' +
+            'onmouseup="SDL.SDLController.seekTracking(event)" ' +
+            '{{bindAttr id="SDL.SDLController.model.style"}} ' +
+            '{{bindAttr disabled="SDL.SDLController.model.disabled"}} ' +
+            '{{bindAttr value="SDL.SDLController.model.valueOfSeekBar"}}/>'+
             '<img class="cd_logo" {{bindAttr class="SDL.SDLController.model.mediaPreset:cd_logoV2"}} {{bindAttr class="SDL.SDLController.model.mode"}} {{bindAttr class="SDL.SDLController.model.isTemplate"}}/>' +
```
Comment (ValeriiMalkov) [inline app/view/media/sdl/controllsView.js]: @ONyzhnyk-Luxoft Add space for elements which related to `input`

File: css/general.css [hunk 5]
```diff
@@ -1777,3 +1777,66 @@ to {
...
+  }
+  #progressBarWithoutSeek::-webkit-slider-thumb {  /* slider in Chrome */
+    -webkit-appearance: none;
+    width:15px;
+    height:15px;
+    border: 1px solid #40310a;
+    border-radius: 2px;
+    background-image: linear-gradient(#40310a, #40310a);
+    box-shadow: -13px 0 #40310a, -26px 0 #40310a, -39px 0 #40310a, -52px 0 #40310a, -65px 0 #40310a, -78px 0 #40310a, -91px 0 #40310a, -104px 0 #40310a, -117px 0 #40310a, -130px 0 #40310a, -143px 0 #40310a, -156px 0 #40310a, -169px 0 #40310a,-182px 0 #40310a, -195px 0 #40310a, -208px 0 #40310a, -221px 0 #40310a, -234px 0 #40310a, -247px 0 #40310a, -260px 0 #40310a, -273px 0 #40310a, -286px 0 #40310a, -299px 0 #40310a, -312px 0 #40310a, -325px 0 #40310a, -338px 0 #40310a, -351px 0 #40310a, -364px 0 #40310a, -377px 0 #40310a, -390px 0 #40310a, -403px 0 #40310a, -416px 0 #40310a, -429px 0 #40310a, -442px 0 #40310a, -455px 0 #40310a;
+  }
```
Comment (ValeriiMalkov) [inline css/general.css]: @ONyzhnyk-Luxoft Add empty line

File: app/model/sdl/MediaModel.js [hunk 6]
```diff
@@ -220,12 +188,9 @@ SDL.SDLMediaModel = SDL.ABSAppModel.extend({
    * Hide/display progress bar call back function
    * which handle a seekBar Boolean parameter
    */
+  disabled: true,
```
Comment (ValeriiMalkov) [inline app/model/sdl/MediaModel.js]: @ONyzhnyk-Luxoft Description which above `disabled` parameter related to the function below ?

File: app/controller/sdl/Abstract/Controller.js (hunk 1 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft Removed

File: app/model/sdl/Abstract/AppModel.js (hunk 2 above)
Comment (ONyzhnyk-Luxoft) [reply app/model/sdl/Abstract/AppModel.js]: @ValeriiMalkov description added

File: app/model/sdl/MediaModel.js (hunk 3 above)
Comment (ONyzhnyk-Luxoft) [reply app/model/sdl/MediaModel.js]: @ValeriiMalkov removed

File: app/model/sdl/MediaModel.js (hunk 6 above)
Comment (ONyzhnyk-Luxoft) [reply app/model/sdl/MediaModel.js]: @ValeriiMalkov fixed

File: css/general.css (hunk 5 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @ValeriiMalkov done

File: app/view/media/sdl/controllsView.js (hunk 4 above)
Comment (ONyzhnyk-Luxoft) [reply app/view/media/sdl/controllsView.js]: @ValeriiMalkov added

File: app/controller/sdl/Abstract/Controller.js [hunk 7]
```diff
@@ -196,17 +196,47 @@ SDL.SDLController = Em.Object.extend(
...
+      var seekBar = SDL.SDLController.model == null ? ""
+        : SDL.SDLController.model.seekBar;
+      if(seekBar === ""){return;}
+      if(seekBar) {
+        SDL.SDLController.model.set('seekBarStyle', "progressBarWithSeek");
+      } else {
+        SDL.SDLController.model.set('seekBarStyle', "progressBarWithoutSeek");
       }
+    }.observes('SDL.SDLController.model.seekBar'),
+    deactivateApp: function() {
```
Comment (BSolonenko) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft  please add empty line and description for function deactivateApp.

File: app/controller/sdl/Abstract/Controller.js [hunk 8]
```diff
@@ -196,17 +196,47 @@ SDL.SDLController = Em.Object.extend(
...
      * StateManager
      */
-    deactivateApp: function() {
-      if (this.model) {
-        if(this.model.seekBar) {
-          this.model.progressBar.remove();
-        }
-        SDL.SDLModel.onDeactivateApp(SDL.States.nextState, this.model.appID);
+    seekBarActive: function() {
+      var seekBar = SDL.SDLController.model == null ? ""
```
Comment (BSolonenko) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft  I prefer something like : 
```JavaScript
var seekBar = SDL.SDLController.model.seekBar;
if(seekBar == null) {
  return;
}
```

File: app/controller/sdl/Abstract/Controller.js [hunk 9]
```diff
@@ -196,17 +196,47 @@ SDL.SDLController = Em.Object.extend(
      * Notification of deactivation of current application model initiated in
      * StateManager
      */
-    deactivateApp: function() {
-      if (this.model) {
-        if(this.model.seekBar) {
-          this.model.progressBar.remove();
-        }
-        SDL.SDLModel.onDeactivateApp(SDL.States.nextState, this.model.appID);
+    seekBarActive: function() {
```
Comment (BSolonenko) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft wrong description, please fix it.

File: app/controller/sdl/Abstract/Controller.js [hunk 10]
```diff
@@ -196,17 +196,47 @@ SDL.SDLController = Em.Object.extend(
...
+  /**
+   * seekTracking function for a tracking progress bar
+   */
+   seekTracking: function(event){
+    if(event.type == 'change'){
+       SDL.SDLController.model.set('pause', true);
+       var percent = event.currentTarget.value;
+       SDL.SDLController.model.set('valueOfSeekBar',percent);
+       if(SDL.SDLController.model.countUp){
+         SDL.SDLController.model.set('currTime',parseInt((SDL.SDLController.model.endTime - SDL.SDLController.model.duration) * percent/100));
```
Comment (BSolonenko) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft  Please calculate the current time in both case first and then call the set function after 'if' operator.

File: app/controller/sdl/Abstract/Controller.js [hunk 11]
```diff
@@ -196,17 +196,47 @@ SDL.SDLController = Em.Object.extend(
...
+       }
+       var number = SDL.SDLController.model.duration + SDL.SDLController.model.currTime;
+       var params = {};
+       params.appID = SDL.SDLController.model.appID;
+       params.hrs = parseInt(number / 3600); // hours
+       params.min = parseInt(number / 60) % 60; // minutes
+       params.sec = number % 60; // seconds
+       FFW.UI.onSeekMediaClockTimerNotification(params);
+     }
+     SDL.SDLController.model.set('pause',false);
```
Comment (BSolonenko) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft  Are you sure that pause false in any case ?

File: css/general.css [hunk 12]
```diff
@@ -1777,3 +1777,67 @@ to {
     width: 50px;
     z-index: 1001;
 }
+
```
Comment (BSolonenko) [inline css/general.css]: @ONyzhnyk-Luxoft  remove one empty line

File: css/general.css [hunk 13]
```diff
@@ -1777,3 +1777,67 @@ to {
...
+    -webkit-appearance: none;
+            appearance: none;
+    padding: 0;
+    border: 1px solid #ccc;
+    border-radius: 5px;
+    box-shadow: inset 0 1px #ccc, inset 0 1px 1px #575555, inset 0 -2px #ccc;
+    background: #fff linear-gradient(#BCBCBC, #fff0f5);
+    overflow: hidden;
+  }
+  #progressBarWithSeek::-moz-range-thumb {  /* slider in Firefox */
```
Comment (BSolonenko) [inline css/general.css]: @ONyzhnyk-Luxoft add empty line

File: css/general.css [hunk 14]
```diff
@@ -1777,3 +1777,67 @@ to {
...
+    border-radius: 2px;
+    box-shadow: -13px 0 #40310a, -26px 0 #40310a, -39px 0 #40310a, -52px 0 #40310a, -65px 0 #40310a, -78px 0 #40310a, -91px 0 #40310a, -104px 0 #40310a, -117px 0 #40310a, -130px 0 #40310a, -143px 0 #40310a, -156px 0 #40310a, -169px 0 #40310a,-182px 0 #40310a, -195px 0 #40310a, -208px 0 #40310a, -221px 0 #40310a, -234px 0 #40310a, -247px 0 #40310a, -260px 0 #40310a, -273px 0 #40310a, -286px 0 #40310a, -299px 0 #40310a, -312px 0 #40310a, -325px 0 #40310a, -338px 0 #40310a, -351px 0 #40310a, -364px 0 #40310a, -377px 0 #40310a, -390px 0 #40310a, -403px 0 #40310a, -416px 0 #40310a, -429px 0 #40310a, -442px 0 #40310a, -455px 0 #40310a;  
+  }
+  #progressBarWithSeek::-moz-range-track {  /* moving line in Firefox */
```
Comment (BSolonenko) [inline css/general.css]: @ONyzhnyk-Luxoft add empty line

File: css/general.css [hunk 15]
```diff
@@ -1777,3 +1777,67 @@ to {
...
+    background: none;
+    border: none;
+  }
+  #progressBarWithSeek::-webkit-slider-thumb {  /* slider in Chrome */
```
Comment (BSolonenko) [inline css/general.css]: @ONyzhnyk-Luxoft add empty line in all place below

File: css/general.css [hunk 16]
```diff
@@ -1777,3 +1777,67 @@ to {
...
+  #progressBarWithoutSeek::-webkit-slider-thumb {  /* slider in Chrome */
+    -webkit-appearance: none;
+    width:15px;
+    height:15px;
+    border: 1px solid #40310a;
+    border-radius: 2px;
+    background-image: linear-gradient(#40310a, #40310a);
+    box-shadow: -13px 0 #40310a, -26px 0 #40310a, -39px 0 #40310a, -52px 0 #40310a, -65px 0 #40310a, -78px 0 #40310a, -91px 0 #40310a, -104px 0 #40310a, -117px 0 #40310a, -130px 0 #40310a, -143px 0 #40310a, -156px 0 #40310a, -169px 0 #40310a,-182px 0 #40310a, -195px 0 #40310a, -208px 0 #40310a, -221px 0 #40310a, -234px 0 #40310a, -247px 0 #40310a, -260px 0 #40310a, -273px 0 #40310a, -286px 0 #40310a, -299px 0 #40310a, -312px 0 #40310a, -325px 0 #40310a, -338px 0 #40310a, -351px 0 #40310a, -364px 0 #40310a, -377px 0 #40310a, -390px 0 #40310a, -403px 0 #40310a, -416px 0 #40310a, -429px 0 #40310a, -442px 0 #40310a, -455px 0 #40310a;
+  }
+
```
Comment (BSolonenko) [inline css/general.css]: @ONyzhnyk-Luxoft remove tabulation

File: app/controller/sdl/Abstract/Controller.js [hunk 17]
```diff
@@ -196,17 +196,47 @@ SDL.SDLController = Em.Object.extend(
...
-        }
-        SDL.SDLModel.onDeactivateApp(SDL.States.nextState, this.model.appID);
+    seekBarActive: function() {
+      var seekBar = SDL.SDLController.model == null ? ""
+        : SDL.SDLController.model.seekBar;
+      if(seekBar === ""){return;}
+      if(seekBar) {
+        SDL.SDLController.model.set('seekBarStyle', "progressBarWithSeek");
+      } else {
+        SDL.SDLController.model.set('seekBarStyle', "progressBarWithoutSeek");
```
Comment (BSolonenko) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft you will never enter here.

File: app/controller/sdl/Abstract/Controller.js (hunk 8 above)
Comment (AKalinich-Luxoft) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft and I would prefer something like 
```
if ( SDL.SDLController.model.seekBar === null) {
  return;
}
var seekBar = SDL.SDLController.model.seekBar;
```

NULL-checking should be done before variable will be actually used

File: app/controller/sdl/Abstract/Controller.js [hunk 18]
```diff
@@ -196,17 +196,47 @@ SDL.SDLController = Em.Object.extend(
...
-        if(this.model.seekBar) {
-          this.model.progressBar.remove();
-        }
-        SDL.SDLModel.onDeactivateApp(SDL.States.nextState, this.model.appID);
+    seekBarActive: function() {
+      var seekBar = SDL.SDLController.model == null ? ""
+        : SDL.SDLController.model.seekBar;
+      if(seekBar === ""){return;}
+      if(seekBar) {
+        SDL.SDLController.model.set('seekBarStyle', "progressBarWithSeek");
```
Comment (AKalinich-Luxoft) [inline app/controller/sdl/Abstract/Controller.js]: @ONyzhnyk-Luxoft 
```
SDL.SDLController.model.set('seekBarStyle', seekBar ? "progressBarWithSeek" : "progressBarWithoutSeek");
```

File: css/general.css [hunk 19]
```diff
@@ -1777,3 +1777,67 @@ to {
...
+    padding: 0;
+    border: 1px solid #ccc;
+    border-radius: 5px;
+    box-shadow: inset 0 1px #ccc, inset 0 1px 1px #575555, inset 0 -2px #ccc;
+    background: #fff linear-gradient(#BCBCBC, #fff0f5);
+    overflow: hidden;
+  }
+  #progressBarWithSeek::-moz-range-thumb {  /* slider in Firefox */
+    border-radius: 2px;
+    box-shadow: -13px 0 #40310a, -26px 0 #40310a, -39px 0 #40310a, -52px 0 #40310a, -65px 0 #40310a, -78px 0 #40310a, -91px 0 #40310a, -104px 0 #40310a, -117px 0 #40310a, -130px 0 #40310a, -143px 0 #40310a, -156px 0 #40310a, -169px 0 #40310a,-182px 0 #40310a, -195px 0 #40310a, -208px 0 #40310a, -221px 0 #40310a, -234px 0 #40310a, -247px 0 #40310a, -260px 0 #40310a, -273px 0 #40310a, -286px 0 #40310a, -299px 0 #40310a, -312px 0 #40310a, -325px 0 #40310a, -338px 0 #40310a, -351px 0 #40310a, -364px 0 #40310a, -377px 0 #40310a, -390px 0 #40310a, -403px 0 #40310a, -416px 0 #40310a, -429px 0 #40310a, -442px 0 #40310a, -455px 0 #40310a;
```
Comment (AKalinich-Luxoft) [inline css/general.css]: @ONyzhnyk-Luxoft please split all these values by three per row like:
```
-13px 0 #40310a,
-26px 0 #40310a,
-39px 0 #40310a, -52px
```
It's quite hard to read and edit

File: css/general.css [hunk 20]
```diff
@@ -1777,3 +1777,67 @@ to {
...
+  #progressBarWithSeek::-webkit-slider-thumb {  /* slider in Chrome */
+    -webkit-appearance: none;
+    width:15px;
+    height:15px;
+    border: 1px solid #818181;
+    border-radius: 2px;
+    background-image: linear-gradient(#dedede, #f2f2f2);
+    box-shadow: -13px 0 #40310a, -26px 0 #40310a, -39px 0 #40310a, -52px 0 #40310a, -65px 0 #40310a, -78px 0 #40310a, -91px 0 #40310a, -104px 0 #40310a, -117px 0 #40310a, -130px 0 #40310a, -143px 0 #40310a, -156px 0 #40310a, -169px 0 #40310a,-182px 0 #40310a, -195px 0 #40310a, -208px 0 #40310a, -221px 0 #40310a, -234px 0 #40310a, -247px 0 #40310a, -260px 0 #40310a, -273px 0 #40310a, -286px 0 #40310a, -299px 0 #40310a, -312px 0 #40310a, -325px 0 #40310a, -338px 0 #40310a, -351px 0 #40310a, -364px 0 #40310a, -377px 0 #40310a, -390px 0 #40310a, -403px 0 #40310a, -416px 0 #40310a, -429px 0 #40310a, -442px 0 #40310a, -455px 0 #40310a;  
+  }
+  #progressBarWithoutSeek{  /* container */
```
Comment (AKalinich-Luxoft) [inline css/general.css]: @ONyzhnyk-Luxoft please add an empty line after every new style block

File: css/general.css [hunk 21]
```diff
@@ -1777,3 +1777,67 @@ to {
...
+    -webkit-appearance: none;
+            appearance: none;
+    padding: 0;
+    border: 1px solid #ccc;
+    border-radius: 5px;
+    box-shadow: inset 0 1px #ccc, inset 0 1px 1px #575555, inset 0 -2px #ccc;
+    background: #fff linear-gradient(#BCBCBC, #fff0f5);
+    overflow: hidden;
+  }
+  #progressBarWithoutSeek::-moz-range-thumb {  /* slider in Firefox */
```
Comment (AKalinich-Luxoft) [inline css/general.css]: @ONyzhnyk-Luxoft officially we are using chromium browser:
https://github.com/smartdevicelink/sdl_hmi#getting-started

So please remove all firefox specific styles

File: app/controller/sdl/Abstract/Controller.js (hunk 7 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @BSolonenko added

File: app/controller/sdl/Abstract/Controller.js (hunk 8 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @BSolonenko fixed

File: app/controller/sdl/Abstract/Controller.js (hunk 9 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @BSolonenko fixed

File: app/controller/sdl/Abstract/Controller.js (hunk 10 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @BSolonenko done

File: app/controller/sdl/Abstract/Controller.js (hunk 11 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @BSolonenko moved in to case

File: css/general.css (hunk 12 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @BSolonenko done

File: css/general.css (hunk 13 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @BSolonenko added

File: css/general.css (hunk 14 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @BSolonenko added

File: css/general.css (hunk 16 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @BSolonenko removed

File: app/controller/sdl/Abstract/Controller.js (hunk 17 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @BSolonenko fixed

File: app/controller/sdl/Abstract/Controller.js (hunk 8 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @AKalinich-Luxoft  fixed

File: app/controller/sdl/Abstract/Controller.js (hunk 18 above)
Comment (ONyzhnyk-Luxoft) [reply app/controller/sdl/Abstract/Controller.js]: @AKalinich-Luxoft  fixed

File: css/general.css (hunk 19 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @AKalinich-Luxoft  fixed

File: css/general.css (hunk 20 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @AKalinich-Luxoft  added

File: css/general.css (hunk 21 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @AKalinich-Luxoft  removed for moz

File: css/general.css (hunk 15 above)
Comment (ONyzhnyk-Luxoft) [reply css/general.css]: @BSolonenko added

**Comment by AKalinich-Luxoft** [comment]
@ONyzhnyk-Luxoft please squash commits before delivery

**Pull request merged**